azure/terraform/KeyVault.rego

```diff
@@ -90,6 +90,7 @@ enableSoftDelete = false {
 }
 
 enableSoftDelete {
+    lower(input.resources[_].type) == "azurerm_key_vault"
     azure_attribute_absence["enableSoftDelete"]
     not azure_issue["enableSoftDelete"]
 }
```

Comment:
@rezoan there is no need for ```lower(input.resources[_].type) == "azurerm_key_vault"```
because ```azure_attribute_absence``` will check that

Reply:
there was a situation where we need to handle 2 types of resource (one parent and one child). for that type of case we need to check the parent type explicitly in calling rule as azure_attribute_absence and azure_issue will check the child type.
But for single resource type you are correct.
Checking the type for single resource would nor hurt but i will remove those checking for single type resoruce in next PR.